frontend/src/components/MapView/Layers/styles.ts

```diff
@@ -1,19 +1,40 @@
 import * as MapboxGL from 'mapbox-gl';
-import { CommonLayerProps } from '../../../config/types';
+import {
+  CommonLayerProps,
+  DataFieldType,
+  PointDataLayerProps,
+} from '../../../config/types';
 import { legendToStops } from './layer-utils';
 
 export const circleLayout: MapboxGL.CircleLayout = { visibility: 'visible' };
-export const circlePaint = (
-  { opacity, legend }: CommonLayerProps,
-  property: string = 'data',
-): MapboxGL.CirclePaint => ({
-  'circle-radius': 8,
-  'circle-opacity': opacity || 0.3,
-  'circle-color': {
-    property,
-    stops: legendToStops(legend),
-  },
-});
+
+export const circlePaint = ({
+  opacity,
+  legend,
+  dataField,
+  dataFieldType,
+}: PointDataLayerProps): MapboxGL.CirclePaint => {
+  const circleColor =
+    dataFieldType === DataFieldType.TEXT
+      ? [
+          'match',
+          ['get', dataField],
+          ...legend
+            .map(legendItem => [legendItem.label, legendItem.color])
+            .reduce((acc, item) => [...acc, ...item]),
```

Comment:
Could we combine this in one step?

Could you add a quick comment with the expected format for the final `MapboxGL.Expression`?